src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -36,6 +36,60 @@ public SimpleInterval(final String contig, final int start, final int end){
         this.end = end;
     }
 
+    /**
+     * Makes an interval by parting the string. The required format is
+     * contig or contig:start or contig:start-end
+     *
+     * example:
+     *
+     * 'chr2', 'chr2:1000000' or 'chr2:1,000,000-2,000,000'
+     */
+    public SimpleInterval(final String string){
+        Locatable loc = parse(string);
```

Comment:
This approach results in our creating TWO `SimpleInterval` objects during construction from a `String`, one of which is thrown away. You should refactor so that `parse()` is an instance method that populates `contig`/`start`/`stop` directly instead of returning a new `Locatable`.